src/internal/recovery/recovery_test.go

```diff
@@ -104,3 +75,113 @@ func TestCreateSnaphot(t *testing.T) {
 		t.Errorf("the file we got out of PFS does not look like a database dump; see debug logs above")
 	}
 }
+
+func TestCreateAndRestoreSnaphot(t *testing.T) {
+	ctx := pctx.TestContext(t)
+	db := dockertestenv.NewMigratedTestDB(t, clusterstate.DesiredClusterState)
+	tracker := track.NewPostgresTracker(db)
+	storage := fileset.NewStorage(fileset.NewPostgresStore(db), tracker, chunk.NewStorage(kv.NewMemStore(), nil, db, tracker))
```

Comment:
We have a test constructor for this.

Reply:
fileset.NewTestStorage only works against an unmigrated database, and can only create tables used by the storage system.  This test wants to write to the `recovery.snapshots` table, but there is no way we can create that in a way that doesn't conflict with NewTestStorage.  (Applying migrations prevents fileset.NewTestStorage from working.)

I definitely get why storage tests don't want all those tables around, but for this "dump the entire database" test, we want the entire database.

(I also like kv.NewMemStore() better than NewFSStore, though not for any real reason.)